main.py

```diff
@@ -1,22 +1,49 @@
 from transformers import AutoTokenizer, AutoModelForCausalLM
 import torch
+import json
+
+#Loads a dataset in the format of the BBQ files and converts it to a list of strings
+def load_BBQ_dataset(dataset_path):
```

Comment:
important: include typehints! 
```suggestion
def load_BBQ_dataset(dataset_path: str) -> list[str]:
```